src/SFA.DAS.Employer.PR.Web.UnitTests/Controllers/YourTrainingProvidersControllerTests.cs

```diff
@@ -135,4 +142,77 @@ long ukprn
         actualPermissionDetails.PermissionToRecruitApprentices.Should().Be(expectedPermissionToRecruitApprenticesText);
         actualPermissionDetails.ChangePermissionsLink.Should().Be("#");
     }
+
+    [Test, MoqInlineAutoData]
+    public void CallsSessionServiceDelete(
+        [Frozen] Mock<IOuterApiClient> outerApiMock,
+       [Frozen] Mock<ISessionService> sessionServiceMock,
+       string employerAccountId,
+       string providerName,
+       long ukprn
+    )
+    {
+        var permission = new Permission { Operations = new List<Operation>(), ProviderName = providerName, Ukprn = ukprn };
+        permission.Operations.Add(Operation.CreateCohort);
+        permission.Operations.Add(Operation.Recruitment);
+
+        var permissions = new List<Permission> { permission };
+        var accountLegalEntities = new List<AccountLegalEntity>
+        {
+            new() {AccountId = 1123,Id=1, Name="account name", PublicHashedId = "12123232", Permissions = permissions}
+        };
+
+        GetEmployerRelationshipsQueryResponse response = new GetEmployerRelationshipsQueryResponse(accountLegalEntities);
+
+        outerApiMock.Setup(o => o.GetAccountLegalEntities(employerAccountId, It.IsAny<CancellationToken>()))
+            .ReturnsAsync(response);
+
+        ClaimsPrincipal user = UsersForTesting.GetUserWithClaims(employerAccountId, EmployerUserRole.Owner);
+        YourTrainingProvidersController sut = new YourTrainingProvidersController(outerApiMock.Object, sessionServiceMock.Object)
+        {
+            ControllerContext = new ControllerContext { HttpContext = new DefaultHttpContext { User = user } }
+        };
+
+        sut.AddUrlHelperMock().AddUrlForRoute(RouteNames.SelectLegalEntity, SelectLegalEntityUrl);
+        var result = sut.Index(employerAccountId, new CancellationToken());
+        sessionServiceMock.Verify(s => s.Delete<AddTrainingProvidersSessionModel>(), Times.Once);
+    }
+
+    [Test, MoqInlineAutoData]
+    public void SetsSelectLegalEntitiesUrl(
+        [Frozen] Mock<IOuterApiClient> outerApiMock,
+     [Frozen] Mock<ISessionService> sessionServiceMock,
+     string employerAccountId,
+     string providerName,
+     long ukprn
+  )
+    {
+        var permission = new Permission { Operations = new List<Operation>(), ProviderName = providerName, Ukprn = ukprn };
+        permission.Operations.Add(Operation.CreateCohort);
+        permission.Operations.Add(Operation.Recruitment);
+
+        var permissions = new List<Permission> { permission };
+        var accountLegalEntities = new List<AccountLegalEntity>
+        {
+            new() {AccountId = 1123,Id=1, Name="account name", PublicHashedId = "12123232", Permissions = permissions}
+        };
```

Comment:
Repetitive code, can create a method to build AccountLegalEntity, it just helps make tests more readable by taking setup noise out of it.

Reply:
done